test/functional/mempool_compatibility.py

```diff
@@ -48,28 +55,44 @@ def run_test(self):
         assert old_tx_hash in old_node.getrawmempool()
         self.stop_node(0)
 
-        self.log.info("Move mempool.dat from old to new node")
+        self.log.info("Copy mempool.dat from old to new node")
         old_node_mempool = os.path.join(old_node.datadir, self.chain, 'mempool.dat')
         new_node_mempool = os.path.join(new_node.datadir, self.chain, 'mempool.dat')
-        os.rename(old_node_mempool, new_node_mempool)
+        shutil.copyfile(old_node_mempool, new_node_mempool)
 
         self.log.info("Start new node and verify mempool contains the tx")
         self.start_node(1)
         assert old_tx_hash in new_node.getrawmempool()
 
         self.log.info("Add unbroadcasted tx to mempool on new node and shutdown")
-        unbroadcasted_tx_hash = new_node.sendtoaddress(recipient, 0.0001)
+        unbroadcasted_tx_hash = new_wallet.send_self_transfer(from_node=new_node)['txid']
         assert unbroadcasted_tx_hash in new_node.getrawmempool()
         mempool = new_node.getrawmempool(True)
         assert mempool[unbroadcasted_tx_hash]['unbroadcast']
         self.stop_node(1)
 
-        self.log.info("Move mempool.dat from new to old node")
-        os.rename(new_node_mempool, old_node_mempool)
+        self.log.info("Copy mempool.dat from new to old node")
+        shutil.copyfile(new_node_mempool, old_node_mempool)
 
-        self.log.info("Start old node again and verify mempool contains both txs")
+        self.log.info("Start old node again and verify mempool does not contain unbroadcasted_tx_hash")
         self.start_node(0, ['-nowallet'])
         assert old_tx_hash in old_node.getrawmempool()
+        # Because the nodes haven't been synced, old_node doesn't have the block that contains the coinbase that unbroadcasted_tx spends.
+        # As coinbases are only valid in a block, and not as loose txns, unbroadcasted_tx won't pass `MemPoolAccept::PreChecks`.
+        assert_raises_rpc_error(-5, "Block not found", old_node.getblock, blockhash=block_hash)
+        assert unbroadcasted_tx_hash not in old_node.getrawmempool()
```

Comment:
I don't understand why this test is being changed. The only change should be replacing the bitcoin core wallet with miniwallet. If you want to do additional changes, they can be done in a separate commit or pull request.